woocommerce/woocommerce
# Fix: attribute query clause for `AND` query type

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR fix the clauses for attribute with `AND` query type to ensure we only return products that have all chosen attributes.



Closes #44789.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the lookup table usage in `WooCommerce > Settings > Products > Advanced > Enable table usage`.
2. Add Color attribute with three options: `blue`, `red`, and `green`.
3. Add two variable products with color attribute terms:
  - Product `Alpha`: `blue` and `green`.
  - Product `Beta`: `green` and `red`.
4. Create variations from those attributes, and ensure all variations have a price and are in stock.
5. On the shop page, using filter widgets/blocks with the `AND` query type or modifying the URL directly, select the `blue` and `green` attributes. See:
  - The loop renders only one product that has both selected attributes.
  - Using the attribute filter block/widget, select the `blue` options, the list is updated correctly to show two options (`blue` and `green`). 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
